core/field_colour.ts

```diff
@@ -584,7 +578,7 @@ export class FieldColour extends Field<string> {
   static fromJson(options: FieldColourFromJsonConfig): FieldColour {
     // `this` might be a subclass of FieldColour if that class doesn't override
     // the static fromJson method.
-    return new this(options['colour'], undefined, options);
+    return new this(options.colour, undefined, options);
```

Comment:
For future reference, the outcome of the discussion was that
- brackets vs dots only matters if we're using ADVANCED_OPTIMIZATIONS  in closure, which we aren't in core. We are doing that in Blockly Games (and some unlucky souls might also be out there doing this for some reason) though
    - even then, it only matters that we are consistently using one or the other within the same object, since we are compiling blockly together with blockly games (so it doesn't matter if the property is renamed)
- if we ever did turn on advanced optimizations, we would need to find a way to convince the compiler not to rename this property in order to use dot notation. but we would also need to convince it not to rename like a billion other things, so presumably we'd have to solve that anyway.

therefore this change is fine.